### PR TITLE
17697: DAE073 - DC SHOP Notice - To EE after they select a plan during Annual Open Enrollment (OE still open - not final confirmation)

### PR DESCRIPTION
**Pull request by krbhavith**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [17697](https://devops.dchbx.org/redmine/issues/17697) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-1 |
| Development Story Points | 2 |
| Peer Review Story Points | 0.5 |
| Ticket Author | Chloe Berger |

### CODE REVIEW

File: app/views/notices/shop_employee_notices/employee_select_plan_during_open_enrollment.html.erb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+</table>
+
+<%= render partial: 'notices/shared/paragraph', :locals =>
+  { content: "Congratulations! You successfully selected <strong>#{notice.plan.plan_name}</strong>." }%>
+
+<%= render partial: 'notices/shared/paragraph', :locals =>
+  { content: "Your employer will need to complete additional steps before your enrollment can be finalized. If your employer successfully completes the final steps in this enrollment process and meets all eligibility requirements, your coverage will go into effect on <strong>#{notice.plan_year.start_on}</strong>." }%>
+
+<%= render partial: 'notices/shared/paragraph', :locals =>
+  { content: "Please note that if your coverage goes into effect on <strong>#{notice.plan_year.start_on}</strong>, your monthly premium contribution will be <strong>#{notice.enrollment.employee_cost}</strong>. Your employer, <strong>#{notice.employer_name}</strong>, will contribute <strong>#{notice.enrollment.responsible_amount}</strong> towards your monthly premium." }%>
```
Comment (Battula) [inline app/views/notices/shop_employee_notices/employee_select_plan_during_open_enrollment.html.erb]: Use number_to_currency view helper to display the costs.
`number_to_currency(notice.enrollment.employee_cost, precision: 2)`

File: app/views/notices/shop_employee_notices/employee_select_plan_during_open_enrollment.html.erb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  { content: "Please note that if your coverage goes into effect on <strong>#{notice.enrollment.effective_on}</strong>, your monthly premium contribution will be <strong>#{number_to_currency(notice.enrollment.employee_cost, precision: 2)}</strong>. Your employer, <strong>#{notice.employer_name}</strong>, will contribute <strong>#{number_to_currency(notice.enrollment.employer_contribution, precision: 2)}</strong> towards your monthly premium." }%>
+
+<%= render partial: 'notices/shared/paragraph_with_margins', :locals =>
+  { content: "If your employer is unable to meet all eligibility requirement and your coverage cannot go into effect, you will receive a notice informing you of next steps.", top_margin: '10px', bottom_margin: '10px'}%>
+
+<%= render partial: 'notices/shared/paragraph_with_headers', :locals =>
+  { contents: [""], heading: 'For Questions or Assistance:', sub_heading: ''} %>
+
+<%= render partial: 'notices/shared/paragraph_with_margins', locals:
+  { content: "You can contact #{Settings.site.short_name} with any questions:"} %>
```
Comment (nks2109) [inline app/views/notices/shop_employee_notices/employee_select_plan_during_open_enrollment.html.erb]: Please add word 'also' after you can

File: app/models/factories/enrollment_renewal_builder.rb [hunk 3]
```diff
@@ -6,14 +6,16 @@ def generate_passive_renewal(options = {})
...
-        save_renewal_enrollment(renewal_enrollment, enrollment)
+        save_renewal_enrollment(renewal_enrollment, enrollment, options)
       end
     end
 
-    def save_renewal_enrollment(renewal_enrollment, active_enrollment)
-      if renewal_enrollment.save
+    def save_renewal_enrollment(renewal_enrollment, active_enrollment, options={})
+      if renewal_enrollment.save # This will generate hbx_id to enrollment
+        renewal_enrollment.send(options.fetch(:aasm_event, :renew_enrollment)) # After save, so that it will handle state machine hooks
```
Comment (saikumar9) [inline app/models/factories/enrollment_renewal_builder.rb]: Triggered notice generation after saving the hbx_enrollment, this helps in finding the **hbx_id** of the saved Enrollment.

File: spec/models/factories/shop_enrollment_renewal_factory_spec.rb [hunk 4]
```diff
@@ -112,6 +112,33 @@
...
+            renewing_employer.renewing_plan_year.update_attribute(:open_enrollment_end_on, TimeKeeper.date_of_record + 1.day)
+            generate_passive_renewal
+          end
+
+          it "should not have nil hbx_id" do
+            queued_job = ActiveJob::Base.queue_adapter.enqueued_jobs.find do |job_info|
+              job_info[:job] == ShopNoticesNotifierJob
+            end
+
+            expect(queued_job[:args][2]["hbx_enrollment_hbx_id"]).not_to eq nil
```
Comment (saikumar9) [inline spec/models/factories/shop_enrollment_renewal_factory_spec.rb]: Good test coverage for verifying the enrollment's **hbx_id.**

**Comment by jayreddy519** [comment]
Created new branch based off MA master..
Making changes on  new PR #1290 to avoid rework.

**Pull request closed without merging**